parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -0,0 +1,396 @@
+                                                      args=(self._kill_event,))
+        self._task_puller_thread.start()
+        self._result_pusher_thread.start()
+
+        start = None
+        abort_flag = False
+
+        logger.info("Loop start")
+
+        # TODO : Add mechanism in this loop to stop the fabric.
+        # This might need a multiprocessing event to signal back.
+        while not abort_flag:
+            time.sleep(0.1)
+
+        for proc_id in self.procs:
+            self.procs[proc_id].terminate()
+
+        # self._task_puller_thread.join()
+        # self._result_pusher_thread.join()
+        delta = time.time() - start
+        logger.info("Fabric ran for {} seconds".format(delta))
+        sys.exit(-1)
+
+
+def execute_task(bufs):
+    """Deserialize the buffer and execute the task.
+
+    Returns the serialized result or exception.
+    """
+    user_ns = locals()
+    user_ns.update({'__builtins__': __builtins__})
+
+    f, args, kwargs = unpack_apply_message(bufs, user_ns, copy=False)
+
+    fname = getattr(f, '__name__', 'f')
+    prefix = "parsl_"
+    fname = prefix + "f"
+    argname = prefix + "args"
+    kwargname = prefix + "kwargs"
+    resultname = prefix + "result"
+
+    user_ns.update({fname: f,
+                    argname: args,
+                    kwargname: kwargs,
+                    resultname: resultname})
+
+    code = "{0} = {1}(*{2}, **{3})".format(resultname, fname,
+                                           argname, kwargname)
+
+    try:
+        logger.debug("[RUNNER] Executing: {0}".format(code))
+        exec(code, user_ns, user_ns)
+
+    except Exception as e:
+        logger.warning("Caught exception; will raise it: {}".format(e))
+        raise e
+
+    else:
+        logger.debug("[RUNNER] Result: {0}".format(user_ns.get(resultname)))
+        return user_ns.get(resultname)
+
+
+def worker(worker_id, task_queue, result_queue, worker_queue):
+    """
+
+    TODO : Add daimyo id to distinguish between multiple daimyo runs under same run
+
+    Put request token into queue
+    Get task from task_queue
+    Pop request from queue
+    Put result into result_queue
+    """
+    start_file_logger('{}/fabric_single_node_{}.log'.format(args.logdir, worker_id),
+                      0,
+                      level=logging.DEBUG if args.debug is True else logging.INFO)
+
+    # Sync worker with master
+    logger.info('Worker ID: {} started'.format(worker_id))
+
+    while True:
+        worker_queue.put(worker_id)
+
+        # The worker will receive {'task_id':<tid>, 'buffer':<buf>}
+        req = task_queue.get()
+        tid = req['task_id']
+        logger.info("Got task : {}".format(tid))
+
+        try:
+            worker_queue.get(worker_id)
+        except queue.Empty:
+            logger.warning("Worker ID: {} failed to remove itself from ready_worker_queue".format(worker_id))
+            pass
```

Comment:
what's the effect of this failure? that we keep requesting tasks for it even though it doesn't need one?

Reply:
I've run into this just once and I don't understand how this exception can be triggered. We put the worker_id into the worker_queue and then take it out once a task is received. So we should be able to get the worker_id out. However if the item can't be removed there's no harm.

Reply:
This entire mechanism has now been removed in favor of the `TASK_PULL_THREAD` just filling up a `ready_task_queue`.

Reply:
I'm most likely going to rollback these changes